location.md

```diff
@@ -0,0 +1,234 @@
+     ```
+    
+* #### List location by UUID.
+
+    Retrieve a location by its UUID. Returns a `404 Not Found` status if location not exists. If user not logged 
+    in to perform this action, a `401 Unauthorized` status returned.
+    
+    ```console
+    GET /location/:target_location_uuid
+    ```
+   
+### Create a location
+
+* To Create a location you need to specify below attributes in the request body.If you are not logged in to perform this action,
+ a `401 Unauthorized` status returned.
+
+    #### Attributes
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *name* | `String` | Name of the location (Required)
+    *address1* | `String` | Address of the location (Required)
+    *description* | `String` | Description
+    *cityVillage* | `String` | City/village
+    *stateProvince* | `String` | State and province
+    *country* | `String` | Country    
+    *postalCode* | `String` | Postal code of the location 
+    *latitude* | `String` | Latitude    
+    *longitude* | `String` | Longitude
+    *countyDistrict* | `String` | Disitrict
+    *tags* | `Array[]: LocationTag UUID` | UUID's of the location tags
+    *parentLocation* | `Parent Location UUID` | UUID of the target parent location
+    *childLocations* | `Array[]: Child Location UUID` | UUID's of the target child locations
+    *attributes* | `Array[]: Attribute UUID` | UUID's of location attributes  
+
+   
+    ```console
+        POST /location
+        {
+          "name": "name ",
+          "description": "string",
+          "address1": "string",
+          "address2": "string",
+          "cityVillage": "string",
+          "stateProvince": "string",
+          "country": "string",
+          "postalCode": "string",
+          "latitude": "string",
+          "longitude": "string",
+          "countyDistrict": "string",
+          "address3": "string",
+          "address4": "string",
+          "address5": "string",
+          "address6": "string",
+          "tags": [
+            "target_location_tag_uuid"
+          ],
+          "parentLocation": "target_parent_location_uuid",
+          "childLocations": [
+            "target_child_location_uuid"
+          ],
+          "attributes": [
+              {
+                 "attributeType": "target_attributeType_uuid",
+                  "value": "value_for_attribute"
+              }
+          ]
+        }
+    ```
+### Update a location
+
+*  Update a target location with given UUID, this method only modifies properties in the request. Returns a `404 Not Found` 
+status if location not exists. If user not logged in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *uuid* | `target_location_uuid` | Target location resource UUID
+    
+    #### Attributes
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *resource* | `Location` | Location resource with updated properties.
```

Comment:
Do I have to include all properties? Or just those that should change? If the latter, then is there a way to clear out an optional property that currently has a value (e.g., `"foobar": null`)?